buildSrc/src/main/groovy/org/logstash/gradle/tooling/SnapshotArtifactURLs.groovy

```diff
@@ -0,0 +1,29 @@
+package org.logstash.gradle.tooling
+
+import groovy.json.JsonSlurper
+
+/**
+ * Helper class to obtain project specific (e.g. Elasticsearch or beats/filebeat) snapshot-DRA artifacts.
+ * We use it in all cases apart from release builds.
+ * */
+class SnapshotArtifactURLs {
+    /**
+     * Returns a list of the package and package SHA(512) URLs for a given project / version / downloadedPackageName
+     * */
+    static def packageUrls(String project, String projectVersion, String downloadedPackageName) {
+        String artifactSnapshotVersionsApiPrefix = "https://artifacts-snapshot.elastic.co"
+
+        // e.g. https://artifacts-snapshot.elastic.co/elasticsearch/latest/8.11.5-SNAPSHOT.json
+        String apiResponse = "${artifactSnapshotVersionsApiPrefix}/${project}/latest/${projectVersion}.json".toURL().text
+        def artifactUrls = new JsonSlurper().parseText(apiResponse)
+        String manifestUrl = artifactUrls["manifest_url"]
+        // e.g. https://artifacts-snapshot.elastic.co/elasticsearch/8.11.5-12345678/manifest-8.11.5-SNAPSHOT.json
+        apiResponse = manifestUrl.toURL().text
+        def packageArtifactUrls = new JsonSlurper().parseText(apiResponse)
+        String packageUrl = packageArtifactUrls["projects"]["${project}"]["packages"]["${downloadedPackageName}.tar.gz"]["url"]
+        String packageShaUrl = packageArtifactUrls["projects"]["${project}"]["packages"]["${downloadedPackageName}.tar.gz"]["sha_url"]
+
+        return [packageUrl, packageShaUrl]
```

Comment:
I would go with a more self explanatory data structure. The clients of this method has to know what to find in first and second positions. I would with something more structured, so that the data can be accessed like:
```groovy
res.packageUrl
res.packageShaUrl
```
instead of 
```groovy
res[0]
res[1]
```
 in order of complexity you can choose:
- Map https://www.baeldung.com/groovy-maps#getItemsFromMap
```groovy
return ["packageUrl": packageUrl, "packageShaUrl":packageShaUrl]
```
- an Expando instance, which generates a class on the fly without the need to declare it https://blog.mrhaki.com/2009/10/groovy-goodness-expando-as-dynamic-bean.html
```groovy
return new Expando(packageUrl: packageUrl, packageShaUrl: packageShaUrl)
```
- plain old POJO, declaring a class with the 2 fields (they can be `public final` because are read only, without pedant getters)

Reply:
Thank you! I used the groovy-maps in 56d5b4287a149691111132b8dbdb8a3a33e7eded